### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by larsmoan**
Detection and classification of skin lesions on the ISIC2017 dataset using YOLOV7 architecture

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Normal

Readme:  Poor
- no insights/discussion/interpretation of results
- no description about how the YOLOv7 algorithm works
- no training/testing loss or accuracy(mAP) plots shown
- no detection results shown in readme (some images are included in the reults folder)

Commit messages:  OK

Code: 
- Only has Jupyter notebooks, does not have the file structure required
- Uses an open source YOLOv7 implementation by someone else, with command line arguments
- No hyperparameter tuning or attempts to improve performance 

Functionality/Performance: 
- Classification accuracy is below 0.8 for all classes
- No detection results shown/No IoU metrics shown

**Comment by larsmoan** [comment]
Thanks for the feedback, obviously a lot of stuff I didn't deliver on during the initial push. I have now done a series of changes, most related to the readme and documentation of the project. I hope this is sufficient.
When it comes to the structure of the project with filenames as model.py, train.pyy etc I find this difficult to implement given the fact that I use the yolov7 library. If I were to follow this rigid structure I would just add more unnecessary code, hope you understand my point of view on this.
Again thanks for the feedback.

**Comment by shakes76** [comment]
# Marking
Some parts marked as per content for due date
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, YOLOv7 external implementation -3
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure not present, needs to be in own folder -2
Shows Usage & Demo & Visualisation & Data usage, no detection results initially shown -2
Module present
Commenting
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good but missing refs -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description minimal -1

**Comment by shakes76** [comment]
Cannot be merged as external implementation of YOLOv7, otherwise grade normally.
